extend/flytekit-xgboost/flytekitplugins/xgboost/task.py

```diff
@@ -114,10 +141,21 @@ def __init__(
         """
         self._config = config
 
+        self._dataset_type = dataset_type
+        inputs = {
+            self._TRAIN_ARG: dataset_type,
+            self._TEST_ARG: dataset_type,
+            self._PARAMS_ARG: XGBoostParameters,
```

Comment:
Does this mean that config has to be compulsorily given?

Reply:
ya, we can make it optional, but even in your case it was compulsory